pkg/api/vlabs/types.go

```diff
@@ -526,6 +526,16 @@ type CustomCloudProfile struct {
 	PortalURL                  string                      `json:"portalURL,omitempty"`
 }
 
+// HasCoreOS returns true if the cluster contains coreos nodes
+func (p *Properties) HasCoreOS() bool {
```

Comment:
you shouldn't need this function in here, the one in `pkg/api/types.go` should be enough, where is this one being used?

Reply:
@CecileRobertMichon It's for addon validation [here](https://github.com/alexeldeib/aks-engine/blob/799d506142256bbf44f76f7147625866d777fa61/pkg/api/vlabs/validate.go#L550-L565) and [here.](https://github.com/alexeldeib/aks-engine/blob/799d506142256bbf44f76f7147625866d777fa61/pkg/api/vlabs/validate.go#L550-L565) I followed the same pattern as `HasWindows()?`

Reply:
got it